File: path.go
package main

import (
  "fmt"
  "path/filepath"
)

func main() {

  // 构建
  pa := filepath.Join("/home/coding/","workspace","mycode","README.md")
  fmt.Println(pa)

  // 拆分
  p,f := filepath.Split(pa)
  fmt.Println(p)
  fmt.Println(f)
  fmt.Println(filepath.Ext(pa));

  // 
 // print(filepath.IsFile(pa)) //无判断path是文件还是目录的函数
 // print(filepath.IsDir(pa)) 
  fmt.Println(filepath.IsAbs(pa)) //
  abss,_ := filepath.Abs(pa)
  fmt.Println(abss) 
  fmt.Println(filepath.Base(pa)) 
  fmt.Println(filepath.Dir(pa)) 
  fmt.Println(filepath.Ext(pa)) 
}

File: path.py
#!/usr/bin/python
# -- coding: utf-8 --

import os

# 构建
pa = os.path.join("/home/coding/","workspace","mycode","README.md")
print(pa)
path = pa


# 拆分
sp = os.path.split(pa)
print(sp)
print(os.path.splitext(pa))
print(os.path.splitext(sp[1]))

# 查询判断
print(os.path.isfile(pa)) #是文件
print(os.path.isdir(pa)) #是目录？
print(os.path.isabs(pa)) #是绝对路径？
print(os.path.abspath(pa)) #返回pa的绝对路径
print(os.path.basename(path)) #返回文件名
print(os.path.dirname(path)) #返回目录路径
print(os.path.exists(path)) #路径存在？

